lib/Backends/OpenCL/OpenCLDeviceManager.h

```diff
@@ -0,0 +1,109 @@
+/**
+ * Copyright (c) 2017-present, Facebook, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#ifndef GLOW_BACKENDS_OPENCL_OPENCLDEVICEMANAGER_H
+#define GLOW_BACKENDS_OPENCL_OPENCLDEVICEMANAGER_H
+
+#include "glow/Backends/QueueBackedDeviceManager.h"
+
+#if defined(__APPLE__) || defined(__MACOSX)
+#include "OpenCL/opencl.h"
+#else
+#include <CL/cl.h>
+#endif
+
+namespace glow {
+namespace runtime {
+
+/// A class that contains an openCL device buffer. It frees the buffer when it
+/// is destroyed. Can be extended to store multiple buffers and rotate through
+/// them. Also tracks number of functions using this buffer.
+class OpenCLBuffer {
+  /// The OpenCL buffer being stored.
+  cl_mem buffer_;
+  /// Count of functions using this buffer.
+  unsigned int users_{0};
```

Comment:
Should the operations on `users_` be atomic to avoid any possible race conditions?

Reply:
Since the OpenCLDeviceManager inherits from queuebackedDeviceManager, all calls are serialized so there shouldn't be any race conditions.